api/src/main/java/org/pmiops/workbench/notebooks/NotebooksConfig.java

```diff
@@ -37,9 +43,21 @@
   @Bean(name = USER_NOTEBOOKS_CLIENT)
   @RequestScope(proxyMode = ScopedProxyMode.DEFAULT)
   public ApiClient notebooksApiClient(
-      UserAuthentication userAuthentication, WorkbenchConfig workbenchConfig) {
+      UserAuthentication userAuthentication, WorkbenchConfig workbenchConfig, HttpHeaders headers) {
     ApiClient apiClient = buildNotebooksApiClient(workbenchConfig);
     apiClient.setAccessToken(userAuthentication.getCredentials());
+
+    // We pass-through the "Referer" header to outgoing Proxy API requests. Leonardo verifies this
+    // Header for all incoming traffic. See IA-2469, RW-6412.
+    Optional<String> referer =
+        Optional.ofNullable(headers.get("Referer"))
+            .filter(Predicates.not(List::isEmpty))
+            .map(h -> h.get(0));
+    if (referer.isPresent()) {
+      apiClient.addDefaultHeader("Referer", referer.get());
+    } else {
+      log.info("no Referer request header found, requests to the Leo proxy API may be rejected");
```

Comment:
something that could be helpful going forward is logging the full request path when there is no Referer present to give AoU a better idea of what requests are failing. we do the same in Leonardo: https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala#L279

Reply:
This is logged in the context of a request, and therefore this is already available to us